pulp_ansible/app/tasks/collections.py

```diff
@@ -431,6 +437,18 @@ async def _add_collection_version(self, api_version, collection_version_url, met
 
         info = metadata["metadata"]
 
+        if self.add_dependents:
+            dependencies = info["dependencies"]
+            for name, version in dependencies.items():
+                requirements_file = RequirementsFileEntry(
+                    name=name,
+                    version=version,
+                    source=None,
+                )
+                if requirements_file not in self.collection_info:
+                    self.collection_queue.put_nowait(requirements_file)
+                    self.collection_info.append(requirements_file)
```

Comment:
I think you also need to check if it is on the queue,
pulp-installer 3.10.0 requirement: ansible.posix
.
.
.
pulp-installer 3.6.0 requirement: ansible.posix

Reply:
I don't think so. As far as I understand async code is still serial and .put_nowait() is not a coroutine, so these two lines should always get executed before another async task is ran.

Reply:
I thought `put_nowait` was a coroutine

Reply:
According to this, https://docs.python.org/3/library/asyncio-queue.html#asyncio.Queue.put `put` is, but `put_nowait` is not.